en/services/command.md

```diff
@@ -8,28 +8,55 @@ These parameters and the command id are encoded in [COMMAND_INT](#COMMAND_INT) o
+This is because it allows the co-ordinate frame to be specified for location and altitude values, which may otherwise be "unspecified".
+In addition latitudes/longitudes can be sent with greater precision in a `COMMAND_INT` as scaled integers in params 5 and 6 (than when sent in float values in `COMMAND_LONG`).
+
+`COMMAND_LONG` must be used for sending `MAV_CMD` commands that send float properties in parameters 5 and 6, as these values would be truncated to integers if sent in `COMMAND_INT`.
 
+Commands that are not positional or that specify integers in params 5 and 6 can be used in either message, if supported by the flight stack.
+
+Flight stacks may support commands in either message `COMMAND_INT` or `COMMAND_LONG` or both, albeit with a loss of precision, rounding errors, and/or undefined frames of reference.
+However they are encouraged to only support positional commands in `COMMAND_INT`, and commands that have float values in param 5 and 6 in `COMMAND_LONG`.
+The flight stack can reject commands sent in the "wrong" message type using the [COMMAND_ACK.result](#COMMAND_ACK) of `MAV_RESULT_COMMAND_LONG_ONLY` or `MAV_RESULT_COMMAND_INT_ONLY`, as appropriate. 
+Flight stacks that only support a particular command in a particular message type can more generally use these result values to indicate the correct message type for a command.
 
 ## Sequences
 
+### COMMAND_INT sequence
+
+[![Mermaid Sequence: Command INT](https://mermaid.ink/img/pako:eNplj90KwjAMhV-l5EphvkAFQTYREfViXhYktJkW1nZ26YWMvbv1ZyCYixDO-XJIBtDBEEjo6Z7Ia6osXiO6pfIiV4eRrbYdehbbsv4Xqxg8feTsL1artyBFeToc1sfqsjueZ03Oo-K1hq6f_8K5S1FzzhNsHYXEH_cdMvlT1LrcQwGOokNr8sXDi1XAN3KkQObRUIOpZQXKjxlNnUGmjbEcIsgG254KwMShfngNkmOiCfp-_aXGJ9ehWtc?type=png)](https://mermaid-js.github.io/mermaid-live-editor/edit#pako:eNplj90KwjAMhV-l5EphvkAFQTYREfViXhYktJkW1nZ26YWMvbv1ZyCYixDO-XJIBtDBEEjo6Z7Ia6osXiO6pfIiV4eRrbYdehbbsv4Xqxg8feTsL1artyBFeToc1sfqsjueZ03Oo-K1hq6f_8K5S1FzzhNsHYXEH_cdMvlT1LrcQwGOokNr8sXDi1XAN3KkQObRUIOpZQXKjxlNnUGmjbEcIsgG254KwMShfngNkmOiCfp-_aXGJ9ehWtc)
+
+The request is sent by the ground station encoded within a [COMMAND_INT](#COMMAND_INT).
+The drone processes the request and responds quickly with a [COMMAND_ACK](#COMMAND_ACK) indicating the result.
+The result is that the command has been accepted (`COMMAND_ACK.result=MAV_RESULT_ACCEPTED`), has been accepted and is in progress (MAV_RESULT_IN_PROGRESS) or it has been rejected with some reason code.
+Note that "accepted" means that the command is valid and the flight stack will attempt to act on it, not that the command has completed.
+Most commands just return "accepted" (they are not implemented as long running commands).
+
+`COMMAND_ACK.result_param2` may also include additional information about the reason for command rejection in a command-specific enum.
+
+If no ACK is received the GCS will resend `COMMAND_INT` for a flight-specific number of times before giving up.
```

Comment:
How is resending done on PX4 for command_int? I.e. unlike COMMAND_LONG there is no confirmation field. Is the same message just resent with no checking, or is the SEQ updated, or what?

Reply:
I don't know. From what I'm aware of nothing is used for neither COMMAND_LONG or COMMAND_INT. The same command is just re-sent until it works, then the next command is sent.

Reply:
Thanks - that's what I'm seeing in QGC. 

It feels a bit unreliable in that if the ack was sent but missed by GCS it will resend. I guess in most cases the vehicle will either reject the new command, or accept it but since it is already in the right state it won't "do" anything. Not sure. 

I'll leave it like this for now. @peterbarker FYI.